### ISSUE TITLE
zap.Error() nil pointer dereference

### ISSUE THREAD

**Issue by knarz** [uber-go/zap#92]
Hello,

`zap.Error()` is a convenience method so I feel like making it handle `nil` errors gracefully seems like a reasonable move.

Example use case: I have an http middleware that logs all requests. The wrapped http handler returns an `error` and then the next line might be something like this:
```go
logger.Info("time to process request", zap.Error(err), zap.Duration("delta", time.Since(t0)), zap.String("path", path))
```

With the current `zap.Error()` this would panic if `err = nil` and require a check for nil to prevent just that.

I'd be happy to submit a PR if this seems like a sane addition.

**Comment by akshayjshah** [comment]
Great point - I'd love a PR and a unit test to cover it!

**Comment by prashantv** [comment]
Fixed by #95.
